### ISSUE TITLE
API to change HP of a creature without triggering Combat Numbers

### ISSUE THREAD

**Issue by apoapostolov** [1000nettles/combat-numbers#21]
The macro below is a quick way for me to give enemies one more turn in case the creature had a sad day and the players owned a hard encounter due to circumstances. It works very well, but now with Combat Numbers, it is exposed to players as healing.

I need a method to update the HP without exposing it visually.

for ( let token of canvas.tokens.controlled ) {
  let HP = token.actor.data.data.attributes.hp.value;
  let Max_HP = token.actor.data.data.attributes.hp.max;
  token.actor.update({
    "data.attributes.hp.value": HP + 5,
    "data.attributes.hp.max" : Max_HP + 5
  });
};

**Comment by 1000nettles** [comment]
Currently, you should be able to hide any HP changes that you make via the Combat Numbers controls / toolbar button: `# > Visibility`.

However, if I'm correct in what you're asking for, it would be ideal to have a way programatically to turn this off / on via a macro. Is that correct?

**Comment by apoapostolov** [comment]
Yes that is a very good solution. I could turn off visibility, modify HP and turn visibility in the same macro flow.

**Comment by 1000nettles** [comment]
Cool, this sounds good and make sense I'll see what I can do.

**Comment by jacobjorvang** [comment]
If doing an API I wonder if it would be possible to display a false number, to account for an NPC's secret damage reduction to certain types of damage (fire resistance etc.)

**Comment by apoapostolov** [comment]
@1000nettles is there a chance to get an API call to turn on and off combat numbers? really needed.

**Comment by 1000nettles** [comment]
@apoapostolov I'm looking to spend some time next weekend working on Combat Numbers and can make sure this is included. I did notice that you forked the module itself - are you working on this feature? Just want to make sure we're not duplicating work.

**Comment by apoapostolov** [comment]
No, it was a misclick actually. :) Not working on it.

**Comment by 1000nettles** [comment]
Finally available now in v1.1.0! Your original ask can be accomplished with:

```javascript
window.combatNumbers.setIsPauseBroadcast(true);
// do your other macro stuff
window.combatNumbers.setIsPauseBroadcast(false);
```
